pandas/core/frame.py

```diff
@@ -6243,13 +6243,28 @@ def explode(self, column: Union[str, Tuple]) -> "DataFrame":
         if not self.columns.is_unique:
             raise ValueError("columns must be unique")
 
-        result = self[column].explode()
-        return (
+        column_with_index = self[column].reset_index()
+        result = (
             self.drop([column], axis=1)
-            .join(result)
-            .reindex(columns=self.columns, copy=False)
+            .reset_index()
+            .join(column_with_index[column].explode())
         )
 
+        if isinstance(self.index, MultiIndex):
+            result.index = pandas.MultiIndex.from_frame(
+                result.iloc[:, : self.index.nlevels]
+            )
+            result.index.rename(
+                [i or None for num, i in enumerate(self.index.names)], inplace=True
```

Comment:
I'm sure I'm missing something, but why can't this be simply `result.index.names = self.index.names`?

Reply:
No reason :) Change pushed, thanks for the suggestion!